### PR TITLE
Fix crash on invalid packet type. Add test

### PR DESCRIPTION
**Pull request by voronin-opensoft**

### CODE REVIEW

File: engineio/socket.py [hunk 1]
```diff
@@ -41,8 +41,10 @@ def poll(self):
 
     def receive(self, pkt):
         """Receive packet from the client."""
+        packet_name = packet.packet_names[pkt.packet_type] \
+            if pkt.packet_type in packet.packet_names else ''
```
Comment (AlexElvers) [inline engineio/socket.py]: This will always return an empty string. `pkt.packet_type` is an int, `packet.packet_names` is a list of strings.

Comment (dimka665) [inline engineio/socket.py]: Yes. Thanks

File: engineio/socket.py [hunk 2]
```diff
@@ -41,8 +41,10 @@ def poll(self):
 
     def receive(self, pkt):
         """Receive packet from the client."""
+        packet_name = packet.packet_names[pkt.packet_type] \
+            if pkt.packet_type <= len(packet.packet_names) else 'UNKNOWN'
```
Comment (miguelgrinberg) [inline engineio/socket.py]: Shouldn't this be "less" instead of "less than"?

**Comment by miguelgrinberg** [comment]
Out of curiosity, what client are you using that feeds invalid packet types to the server?

**Comment by voronin-opensoft** [comment]
Hello.
It fixes crash on invalid packet type. It solves problem from #30 issue . We uses `socket.io-client: 1.4.5` and gets WebSocket packet consist of one byte with random value: https://github.com/miguelgrinberg/python-engineio/blob/master/engineio/socket.py#L190

**Comment by miguelgrinberg** [comment]
Thanks!

**Pull request merged**